### PR TITLE
[feature] Spurious session creation

### PR DESCRIPTION
**Pull request by lyndsysimon**

### CODE REVIEW

**Comment by lyndsysimon** [comment]
@sloria - can we get this included on 0.39 or 0.40?

**Comment by lbanner** [comment]
@sloria @icereval .... long existent issue/challenge as far as I am aware. Can we get this reviewed and merged?

**Comment by lyndsysimon** [comment]
I just rebased this to current develop. As soon as tests pass, it should be a painless merge.

File: framework/sessions/__init__.py [hunk 1]
```diff
@@ -94,9 +94,16 @@ def set_previous_url(url=None):
     url = url or request.path
     if any([rule(url) for rule in settings.SESSION_HISTORY_IGNORE_RULES]):
         return
+
```
Comment (icereval) [inline framework/sessions/__init__.py]: You should be able to simply remove `set_previous_url` (including where it is used) and `goback`, searching around the code base I don't see where they provide any functionality, I'm assuming I simply missed these when we moved authentication over to CAS.

File: framework/sessions/__init__.py [hunk 2]
```diff
@@ -94,9 +94,16 @@ def set_previous_url(url=None):
...
+    # Do nothing if this is a duplicate of the last visited URL
+    if len(session.data['history']) and url == session.data['history'][-1]:
+        return
+
     session.data['history'].append(url)
     while len(session.data['history']) > settings.SESSION_HISTORY_LENGTH:
         session.data['history'].pop(0)
+    if session.persist:
```
Comment (icereval) [inline framework/sessions/__init__.py]: Would move down to `after_request`

**Comment by sloria** [comment]
@lyndsysimon Once @icereval 's comments are addressed, this is good to merge.

File: framework/sessions/__init__.py (hunk 1 above)
Comment (lyndsysimon) [inline framework/sessions/__init__.py]: `goback()` was indeed never used. I did an exhaustive site-wide search, and the `Session.data['history']` list was also never used. I've removed `goback()`, `set_previous_url()`, the `history` key in the `session.data` dict, and the `after_request` handler.

In total, your comment justified my removing a new 58 lines of OSF code - you are officially my favorite person today. :joy:

File: framework/sessions/__init__.py (hunk 2 above)
Comment (lyndsysimon) [inline framework/sessions/__init__.py]: This method is no longer necessary; see above.

File: framework/sessions/model.py [hunk 3]
```diff
@@ -13,8 +13,7 @@ class Session(StoredObject):
     date_modified = fields.DateTimeField(auto_now=True)
     data = fields.DictionaryField()
 
-    def __init__(self, *args, **kwargs):
-        super(Session, self).__init__(*args, **kwargs)
-        # Initialize history to empty list if not found
-        if 'history' not in self.data:
-            self.data['history'] = []
+    @property
+    def persist(self):
```
Comment (sloria) [inline framework/sessions/model.py]: This doesn't seem to be used anywhere; @lyndsysimon Can this be removed?

Comment (lyndsysimon) [inline framework/sessions/model.py]: Yep, thanks for the catch. Done in d70a3ce.

**Pull request merged**